pkg/client/connector/k8s_config.go

```diff
@@ -1,17 +1,50 @@
 package connector
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
 	"sort"
 
 	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/rest"
 
 	"github.com/datawire/ambassador/pkg/kates"
+	"github.com/telepresenceio/telepresence/v2/pkg/iputil"
 )
 
+// The dns config part of the kubeconfigExtension that is read from the kubeconfig
+type dns struct {
+	// LocalIP is the address of the local DNS server. This entry is only
+	// used on Linux system that are not configured to use systemd.resolved and
+	// can be overridden by using the option --dns on the command line and defaults
+	// to the first line of /etc/resolv.conf
+	LocalIP iputil.IPKey `json:"local-ip,omitempty"`
+
+	// RemoteIP is the address of the cluster's DNS service. It will default
+	// to the IP of the kube-dns.kube-system or the dns-default.openshift-dns service.
+	RemoteIP iputil.IPKey `json:"remote-ip,omitempty"`
+
+	// ExcludeSuffixes are suffixes for which the DNS resolver will always fail (or fallback
+	// in case of the overriding resolver).
+	ExcludeSuffixes []string `json:"exclude-suffixes,omitempty"`
+
+	// IncludeSuffixes are suffixes for which the DNS resolver will always attempt to do
+	// a lookup. Includes have higher priority than excludes.
+	IncludeSuffixes []string `json:"include-suffixes,omitempty"`
+
+	// The maximum number of seconds to wait for a cluster side host lookup.
+	LookupTimeout int32 `json:"lookup-timeout,omitempty"`
```

Comment:
Can we use a [`metav1.Duration`](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Duration) instead of an int, so that it doesn't need to be a whole-number of seconds?

Reply:
I thought about that but I'm concerned that users will forget the 's' and run into problems that may be difficult to understand. Do you think it makes sense to configure this timeout in fractions of seconds?

Reply:
Thought about it some more. Duration it is.